Source/Scene/Model.js

```diff
@@ -511,12 +514,20 @@ define([
          */
         this.colorBlendAmount = defaultValue(options.colorBlendAmount, 0.5);
 
+        this.colorShadingEnabled = isColorShadingEnabled(this);
+
+        this._clippingPlanes = undefined;
+        this.clippingPlanes = options.clippingPlanes;
+        // Used for checking if shaders need to be regenerated due to clipping plane changes.
+        this._clippingPlanesState = 0;
+
         /**
-         * The {@link ClippingPlaneCollection} used to selectively disable rendering the model. Clipping planes are not currently supported in Internet Explorer.
+         * Flag for indicating to ModelInstanceCollection that shaders were changed, necessitating updates.
          *
-         * @type {ClippingPlaneCollection}
+         * @type {Boolean}
+         * @private
          */
-        this.clippingPlanes = options.clippingPlanes;
+        this.changedShadersOnLastUpdate = false;
```

Comment:
This flag feels wrong to have. Could `ModelInstanceCollection` instead look to see if model commands are dirty?